### PR TITLE
MEN-5124: add support for `Name <email@example.com>` mail sender/recipients

### PR DESCRIPTION
**Pull request by tranchitella**
Changelog: title

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

File: app/worker/smtp.go [hunk 1]
```diff
@@ -142,3 +147,11 @@ func processSMTPTask(smtpTask *model.SMTPTask, job *model.Job,
 
 	return result, nil
 }
+
+func getEmailAddress(from string) string {
+	emailAddress, err := mail.ParseAddressList(from)
+	if err != nil || len(emailAddress) < 1 {
```
Comment (merlin-northern) [inline app/worker/smtp.go:153]: is it possible that `emailAddress[0].Address` is empty? (if so, I would add a check as well.

Comment (tranchitella) [reply app/worker/smtp.go:153]: No, it is not possible (it would return an error)

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

**Comment by tranchitella** [comment]
The pipeline is green, no idea why `test:integration:prebuilt_client:open_source/master` is not reported back here.

File: app/worker/smtp.go [hunk 2]
```diff
@@ -129,8 +130,12 @@ func processSMTPTask(smtpTask *model.SMTPTask, job *model.Job,
 		}
 	}
 
-	err = smtpClient.SendMail(smtpHostname, auth, from, recipients, msgBuffer.Bytes())
-	l.Debugf("processSMTPTask: smtpClient.SendMail returned %v", err)
```
Comment (merlin-northern) [inline app/worker/smtp.go:133]: perhaps lets keep that debug, just in case?

File: app/worker/smtp.go [hunk 3]
```diff
@@ -129,8 +130,12 @@ func processSMTPTask(smtpTask *model.SMTPTask, job *model.Job,
...
+	fromAddress := getEmailAddress(from)
+	recipientAddresses := make([]string, len(recipients))
+	for i, recipient := range recipients {
+		recipientAddresses[i] = getEmailAddress(recipient)
+	}
+	err = smtpClient.SendMail(smtpHostname, auth, fromAddress, recipientAddresses, msgBuffer.Bytes())
```
Comment (merlin-northern) [inline app/worker/smtp.go:138]: feel free to ignore: perhaps we should check if the `len(recipientAddresses)>0` ?

**Review by merlin-northern — APPROVED** [review]
lgtm, left some minor comments.

File: app/worker/smtp.go (hunk 3 above)
Comment (tranchitella) [reply app/worker/smtp.go]: https://github.com/mendersoftware/workflows/pull/169/commits/aeec87d906a92c8f840d8a627803640ed45e50c6

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

**Pull request merged**